fixtures/swift-codable/tests/bindings/test_codable.kts

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import uniffi.codable_test.*
+
+val simpleRecord = SimpleRecord(
+    string = "Test",
+    boolean = true,
+    integer = 1,
+    floatVar = 1.1,
+    vec = listOf(true),
+)
+
+val jsonSimpleRecord = Json.encodeToString(simpleRecord)
+val deserializedSimpleRecord = Json.decodeFromString<SimpleRecord>(jsonSimpleRecord)
+assert(deserializedSimpleRecord.string == "Test")
+assert(deserializedSimpleRecord.boolean)
+assert(deserializedSimpleRecord.integer == 1)
+assert(deserializedSimpleRecord.floatVar == 1.1)
+assert(deserializedSimpleRecord.vec == listOf(true))
+
+// MultiLayerRecord
+val multilayer = MultiLayerRecord(
+    simpleEnum = SimpleEnum.ONE,
+    reprU8 = ReprU8.TWO,
+    simpleRecord = simpleRecord,
+)
+
+val jsonMultiLayerRecord = Json.encodeToString(multilayer)
+val deserializedMultiLayerRecord = Json.decodeFromString<MultiLayerRecord>(jsonMultiLayerRecord)
+assert(deserializedMultiLayerRecord.simpleEnum == SimpleEnum.ONE)
+assert(deserializedMultiLayerRecord.reprU8 == ReprU8.TWO)
+assert(deserializedMultiLayerRecord.simpleRecord == simpleRecord)
+
+// RecordWithOptionals
+val optionals = RecordWithOptionals(
+    string = null,
+    boolean = null,
+    integer = null,
+    floatVar = null,
+    vec = listOf("A", "B"),
+)
+
+val implicitNulls = Json { explicitNulls = false }
+val jsonRecordWithOptionals = implicitNulls.encodeToString(optionals)
+assert(jsonRecordWithOptionals == """{"vec":["A","B"]}""")
+
+// ComplexEnum
+val withClassDiscriminator = Json { classDiscriminator = "#class" }
```

Comment:
This will embed `"#class": "ComplexEnum.String"` into the resulting JSON string, but some may want to customize the value of that property as well. We can provide this via the `@kotlinx.serialization.SerialName` annotation, but I'm not sure there is an elegant way to customize it via the configuration file. I'm thinking of something like the custom type configuration, e.g.:

```toml
[bindings.kotlin.serialization.ComplexEnum.String]
serial_name = "complex_enum_string"
```

which will put `"#class": "complex_enum_string"` to the resulting string.

Reply:
Assuming you don't intend blocking this PR on that tweak, it might be worth opening a PR where we can discuss this, then add a short comment in the fixture itself which notes what you said and links to the PR, so people exploring this fixture also discovers this.

Re the idea for customizing, that seems good - another alternative might be a format similar to what we did for renaming - https://mozilla.github.io/uniffi-rs/next/renaming.html - which would be more like:

```
[bindings.kotlin.serialization.serial_names]
ComplexEnum.String = "complex_enum_string"
```

(I've no idea if that example makes actual sense in this context, but some degree of consistency makes some sense)